lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -2,47 +2,54 @@ module ActiveModel
   class Serializer
     module Adapter
       class Attributes < Base
+        attr_accessor :_include_tree
+
         def initialize(serializer, options = {})
           super
-          @include_tree = IncludeTree.from_include_args(options[:include] || '*')
+          self._include_tree = IncludeTree.from_include_args(options[:include] || '*')
         end
 
-        def serializable_hash(options = nil)
+        def serializable_hash(options = {})
           options ||= {}
+
           if serializer.respond_to?(:each)
-            result = serializer.map { |s| Attributes.new(s, instance_options).serializable_hash(options) }
+            serializable_hash_for_collection(options)
           else
-            hash = {}
+            serializable_hash_for_single_resource(options)
+          end
+        end
 
-            core = cache_check(serializer) do
-              serializer.attributes(options)
-            end
+        def serializable_hash_for_collection(options = {})
```

Comment:
This should be private, and there probably is no necessity for a default value for `options`.